fastlane/swift/LaneFileProtocol.swift

```diff
@@ -136,13 +131,19 @@ open class LaneFile: NSObject, LaneFileProtocol {
         }
 
         // Call all methods that need to be called before we start calling lanes.
-        fastfileInstance.setUpAllTheThings()
+        let beforeAllSelector = NSSelectorFromString("beforeAllWithCurrentLane:")
+        if fastfileInstance.responds(to: beforeAllSelector) {
+            _ = fastfileInstance.perform(beforeAllSelector, with: laneMethod)
+        }
```

Comment:
Quick question! What is the difference between calling the method directly and calling it via `NSSelectorFromString`? 🤔  TBH, calling methods from a string always scares me since there isn't any compile time check 🤷‍♂️

Reply:
@joshdholtz I am also very much unaware about the way this was initially built, but my guess is that somehow changing the type of the `fastfileInstance` from `Fastfile` to `Lanefile` made calling the method directly to be executing the default implementation of the `LaneFileProtocol`'s extension, instead of the actual implementation declared in the `Fastfile`. 🤔 

<img width="446" alt="image" src="https://user-images.githubusercontent.com/9969006/94810504-62d59480-03f4-11eb-847e-46110737ba5a.png"> 

I also don't like this solution very much and I would love to hear suggestions, as long as it allows `Fastlane.Swift` users to be able to update from `2.158.0` to `2.161.0` without risking to get more regression issues. 🚀 

Unfortunately my time is also limited, though I'll try to unlock some more and play around with the type of `fastfileInstance` to see if my assumption was correct. 👍

Reply:
Yeah, that change was to make spm work properly, but I clearly messed up. It could be what you say, that's why I recommended to change the no-op to something else to see the caller. Also maybe call that method has to be deferent for SPM or Xcode project.

Reply:
@minuscorp we could also try type-casting `fastfileInstance` to `Fastfile` when using an Xcode project, maybe? 🤔

Reply:
```swift
#if !SWIFT_PACKAGE
(fastfileInstance as? Fastfile)?.beforeAll()
#endif
```

Reply:
And in the case of SPM? Did you test if it was working?

Reply:
As I said, in currently resting from two weeks at the hospital, I cannot use a laptop yet sorry. In SPM it should work because the Lanefile comes from the parent dependency, but if an issue is there I'll check it out ASAP I'm healthier 😅

Reply:
@minuscorp no worries, that wasn't my question actually: I was asking if you tested the `beforeAll` & `afterAll` use case at the time you made the PR, wasn't expecting you to test that now. 👌
Get well soon! 🤒

Reply:
IIRC correctly it was working correctly on from SPM, I will revisit it though, although there user base is much smaller. In a future maybe we only have spm, who knows.

Reply:
Alright, I might try testing it on SPM as well later 👍